### PR TITLE
Cleaned up some typing in GridAxis

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Missing time types and any casts

Edit: PR mutated into a general time unit typing cleanup.

### CODE REVIEW

File: ts/Core/Time.ts [hunk 1]
```diff
@@ -898,49 +898,52 @@ class Time {
...
 
         let format: string|undefined,
-            n,
-            lastN = 'millisecond'; // for sub-millisecond data, #4223
+            n: Time.TimeUnit|undefined,
+            // for sub-millisecond data, #4223
+            lastN: Time.TimeUnit = 'millisecond';
 
         for (n in timeUnits) { // eslint-disable-line guard-for-in
+            if (n) {
```
Comment (bre1470) [inline ts/Core/Time.ts:909]: In my opinion the test is not necessary when it comes to keys.

File: ts/Core/Time.ts [hunk 2]
```diff
@@ -954,6 +957,21 @@ class Time {
...
+        to?: string;
+    }
+    export type DateTimeLabelFormatOption = (
+        string|
+        Array<string>|
+        Time.DateTimeLabelFormatObject
+    );
+    export type DateTimeLabelFormatsOption = (
+        Record<TimeUnit, DateTimeLabelFormatOption>
+    );
```
Comment (bre1470) [inline ts/Core/Time.ts:974]: The naming could be simplified as the Time class has a DTL function, so DTLFormatObject etc.

Comment (TorsteinHonsi) [reply ts/Core/Time.ts:974]: I don't know... The name of the internal `resolveDTLFormat` function is an attempt to save filesize, but for the typing I think we should spell it out.

**Review by bre1470 — APPROVED** [review]
Looks good! 👍

**Pull request merged**